java/src/main/native/src/TableJni.cpp

```diff
@@ -563,6 +563,42 @@ bool valid_window_parameters(native_jintArray const &values,
          values.size() == preceding.size() && values.size() == following.size();
 }
 
+static void build_one_column_metadata(JNIEnv *env, jobject meta_obj,
+                                      jfieldID name_id,
+                                      jmethodID children_mid,
+                                      cudf::column_metadata& out) {
+  // get column name
+  cudf::jni::native_jstring col_name(env, (jstring)env->GetObjectField(meta_obj, name_id));
```

Comment:
There's a couple of issues here.  First, it doesn't check for any exceptions that occur from the JVM callback.  Second, we may consider avoiding doing it this way. This performs multiple JVM callbacks with object reflection on each field which would be nice to avoid.

In other places we are handling mapping metadata to columns in a nested structure by "flat-mapping" the metadata. For example, see how `ParquetWriterOptions` handles decimal precision values.  Can we not use a similar approach here?  That would preclude the need to callback into the JVM for every name.

Reply:
This case should be different from the case handing decimal precision values. The native `to_arrow ` requires to rebuild the nested structure of the column metadata. Flattening them will lose the structure info. 
Not sure there would be a better way to avoid callback into the JVM. Will try.

Reply:
Instead of callbacks into JVM, updated to use regular JNI calls to build the native column metadata.

Reply:
good to know. Thanks.